src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/filingpackage/service/filingPackageServiceImpl/GetSubmissionDocumentTest.java

```diff
@@ -0,0 +1,110 @@
+
+        sut = new FilingPackageServiceImpl(efilingReviewServiceMock, accountServiceMock, new FilingPackageMapperImpl());
+    }
+
+
+    @Test
+    @DisplayName("Ok: a submitted document was returned")
+    public void withValidRequestReturnDocument() {
+
+        Mockito.when(efilingReviewServiceMock.findStatusByPackage(ArgumentMatchers.any())).thenReturn(Optional.of(TestHelpers.createFilingPackage()));
+
+        Mockito.when(efilingReviewServiceMock.getSubmittedDocument(Mockito.any(), ArgumentMatchers.eq(TestHelpers.DOCUMENT_ID))).thenReturn(Optional.of(DOC_DATA));
+
+        Optional<SubmittedDocument> result = sut.getSubmittedDocument(TestHelpers.CASE_1_STRING, BigDecimal.ONE, TestHelpers.DOCUMENT_ID);
+
+        Assertions.assertTrue(result.isPresent());
+        Assertions.assertEquals(TestHelpers.NAME, result.get().getName());
+        Assertions.assertEquals(DOC_DATA, result.get().getData());
+
+    }
+
+    @Test
+    @DisplayName("Not found: missing account")
+    public void withValidRequestButMissingAccountReturnEmpty() {
+
+        Optional<SubmittedDocument> result = sut.getSubmittedDocument(TestHelpers.CASE_2_STRING, BigDecimal.ONE, TestHelpers.DOCUMENT_ID);
+
+        Assertions.assertFalse(result.isPresent());
+
+    }
+
+
+    @Test
+    @DisplayName("Not found: no filing package")
+    public void withValidRequestButMissingPackageReturnEmpty() {
+
+        Mockito.when(efilingReviewServiceMock.findStatusByPackage(ArgumentMatchers.any())).thenReturn(Optional.empty());
+
+        Optional<SubmittedDocument> result = sut.getSubmittedDocument(TestHelpers.CASE_1_STRING, BigDecimal.TEN, TestHelpers.DOCUMENT_ID);
+
+        Assertions.assertFalse(result.isPresent());
+
+    }
+
+    @Test
+    @DisplayName("Not found: document not in filing package")
+    public void withValidRequestButMissingDocumentInPackageReturnEmpty() {
+
+        Mockito.when(efilingReviewServiceMock.findStatusByPackage(ArgumentMatchers.any())).thenReturn(Optional.of(TestHelpers.createFilingPackage()));
+
+        Optional<SubmittedDocument> result = sut.getSubmittedDocument(TestHelpers.CASE_1_STRING, BigDecimal.ONE, DOCUMENT_NOT_FOUND);
+
+        Assertions.assertFalse(result.isPresent());
+
+    }
+
+    @Test
+    @DisplayName("Not found: no document")
+    public void withValidRequestButMissingDocumentReturnEmpty() {
+
+        Mockito.when(efilingReviewServiceMock.findStatusByPackage(ArgumentMatchers.any())).thenReturn(Optional.of(TestHelpers.createFilingPackage()));
+
+        Mockito.when(efilingReviewServiceMock.getSubmittedDocument(Mockito.any(), ArgumentMatchers.eq(TestHelpers.DOCUMENT_ID))).thenReturn(Optional.empty());
+
+        Optional<SubmittedDocument> result = sut.getSubmittedDocument(TestHelpers.CASE_1_STRING, BigDecimal.TEN, TestHelpers.DOCUMENT_ID);
+
+        Assertions.assertFalse(result.isPresent());
+
+    }
```

Comment:
Add a test with multiple documents in the filingStatusPackage response but 2nd documents in that list is the actual doc